src/Components/Form/Form.js

```diff
@@ -23,14 +22,14 @@ export class Form extends Component {
     this.setState({currentAnswers: [...this.state.currentAnswers, event.target.textContent]})
   }
 
-  updateAllAnswers = (event) => {
+  updateAllAnswers = async (event) => {
     event.preventDefault();
-    this.setState({allAnswers: [...this.state.allAnswers, this.state.currentAnswers], currentAnswers: []})
+    await this.setState({allAnswers: [...this.state.allAnswers, this.state.currentAnswers], currentAnswers: []})
     if (this.state.allAnswers.length === 1) {
-      this.props.setActivities(this.state.currentAnswers);
-      let relevantQuestions = this.state.currentAnswers.reduce((relevantQuestions, activity) => {
+      this.props.setActivities(this.state.allAnswers[0]);
+      let relevantQuestions = this.state.allAnswers[0].reduce((relevantQuestions, activity) => {
```

Comment:
Good call changing this to allAnswers since we added await to line 27, current answers will always ever be an empty array at this point!

Reply:
Thanks for all the hard work you put into this problem, I am glad it paid off in the end!